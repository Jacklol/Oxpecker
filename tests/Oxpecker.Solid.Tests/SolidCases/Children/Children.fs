module Oxpecker.Solid.Tests.Cases.Children

open System.Runtime.CompilerServices
open Oxpecker.Solid

[<SolidComponent>]
let Component (hello: string) (children: #HtmlElement) =
    h1() {
        hello
        children
    }

// [<SolidComponent>]
// let Test () =
//     div() {
//         Component "xxx" (__() {
//             "World"
//         })
//     }
