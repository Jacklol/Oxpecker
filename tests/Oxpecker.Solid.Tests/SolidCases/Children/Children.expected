import { some } from "./fable_modules/fable-library-js.4.21.0/Option.js";
import { createSignal } from "solid-js";

export function Child(getText) {
    return <h1 onClick={(_arg) => {
            console.log(some(getText()));
        }}>
        {getText()}
    </h1>;
}

export function Parent() {
    const patternInput = createSignal("Hello");
    return <div>
        {Child(patternInput[0])}
    </div>;
}

